### ISSUE TITLE
imghdr used in cover will be removed in python 3.13

### ISSUE THREAD

**Issue by clach04** [kevinhendricks/KindleUnpack#44]
https://github.com/kevinhendricks/KindleUnpack/blob/master/lib/mobi_cover.py#L11

From https://docs.python.org/3/library/imghdr.html

> Deprecated since version 3.11, will be removed in version 3.13: The imghdr module is deprecated (see PEP 594 for details and alternatives).

**Comment by clach04** [comment]
None of the suggested replacements in the PEP support python 2. Could adopt imghdr into projector add additional checks to existing code for types we care about.

I do have a project https://github.com/clach04/magic-fork which might be suitable .

**Comment by kevinhendricks** [comment]
We will keep it in mind but we still have plenty of time since it was just deprecated in 3.11.

filetype might be an option.

We may decide to remove python2 support since all work on kfx input seems to be python3 only.

**Comment by kevinhendricks** [comment]
Also, since the current python lib imghdr.py is a simple self-contained module we may just decide to include a version of it as part of kindleunpack.

**Comment by NebularNerd** [comment]
Another alternative is [PureMagic](https://github.com/cdgriffith/puremagic), imghdr support was recently built into it. You should be able to use it as a drop-in replacement without modification.

**Comment by dougmassay** [comment]
This makes more sense to me upstream (here) than me n my plugin. If we handle the imghdr issue here, there'll be no issue downstream. It seems as though an imghdr-less Python 3.13 is now here.

https://github.com/dougmassay/kindleunpack-calibre-plugin/pull/7

**Comment by kevinhendricks** [comment]
Agreed.  I will commit imghdr.py from Python 3.10 (prior to its deprecation) here so it is available.

Probably tomorrow after making sure its license is compatible.

**Comment by kevinhendricks** [comment]
Extracted imghdr.py from Python 3.10 (CPython Github) and edited it to add the Python license information to the file itself.  No other changes were made.

It was added to the lib folder.

So importing imghdr should now work in a backwards compatible manner for all platforms.
